py/objstringio.c

```diff
@@ -98,7 +98,7 @@ STATIC mp_uint_t stringio_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_t arg,
     switch (request) {
         case MP_STREAM_SEEK: {
             struct mp_stream_seek_t *s = (struct mp_stream_seek_t*)arg;
-            mp_uint_t ref = 0;
+            mp_uint_t new_pos, ref = 0;
```

Comment:
Please minimize changes by avoiding patching this line, and C99-defining new_pos where it's first set.